pandas/tseries/tests/test_converter.py

```diff
@@ -62,6 +62,21 @@ def test_conversion(self):
             np_datetime64_compat('2012-01-02 00:00:00+0000')]), None, None)
         self.assertEqual(rs[0], xp)
 
+        # we have a tz-aware date (constructed to that when we turn to utc it
+        # is the same as our sample)
+        ts = (Timestamp('2012-01-01')
+              .tz_localize('UTC')
+              .tz_convert('US/Eastern')
+              )
+        rs = self.dtc.convert(ts, None, None)
+        self.assertEqual(rs, xp)
+
+        rs = self.dtc.convert(Index([ts - Day(1), ts]), None, None)
+        self.assertEqual(rs[1], xp)
+
+        rs = self.dtc.convert(ts, None, None)
+        self.assertEqual(rs, xp)
```

Comment:
Can you also add  similar test for `Index([ts - Day(1), ts]).to_pydatetime` to test tz aware datetime.datetime objects (as this is actually what was raising the tests I think, as `_mpl_repr()` does return that).